ca/django_ca/managers.py

```diff
@@ -336,9 +342,21 @@ def init(
         # NOTE: Already verified by KeySizeAction, so these checks are only for when the Python API is used
         #       directly.
         generate_key_args = validate_key_parameters(key_size, key_type, ecc_curve)
-        algorithm = parse_hash_algorithm(algorithm)
+        if openssh_ca:
+            algorithm = parse_hash_algorithm(algorithm)
+        else:
+            algorithm = None
         expires = parse_expires(expires)
 
+        if openssh_ca and parent:
+            raise ValueError("OpenSSH does not support intermediate authorities")
+        if not openssh_ca and key_type == "EdDSA":
+            raise ValueError("EdDSA only supported for OpenSSH authorities")
```

Comment:
Q: Is that your restriction or a technical restriction?

Reply:
http://cvsweb.openbsd.org/cgi-bin/cvsweb/src/usr.bin/ssh/PROTOCOL.certkeys?rev=1.8

> The valid key types for CA keys are ssh-rsa, ssh-dss and the ECDSA types
ecdsa-sha2-nistp256, ecdsa-sha2-nistp384, ecdsa-sha2-nistp521. "Chained"
certificates, where the signature key type is a certificate type itself
are NOT supported.